src/main/java/org/truffleruby/language/loader/RequireNode.java

```diff
@@ -130,10 +130,37 @@ protected boolean require(VirtualFrame frame, String feature,
                     featureLoader.ensureCExtImplementationLoaded(feature, callNode);
 
                     if (getContext().getOptions().CEXTS_LOG_LOAD) {
-                        Log.info("loading cext module %s", expandedPath);
+                        Log.info("loading cext module %s (requested as %s)", expandedPath, feature);
+                    }
+
+                    final CallTarget callTarget;
+
+                    try {
+                        callTarget = featureLoader.parseSource(source);
+                    } catch (JavaException e) {
+                        if (e.getCause() instanceof RuntimeException
+                                && e.getCause().getCause() instanceof IOException
+                                && e.getCause().getCause().getCause() instanceof UnsatisfiedLinkError) {
```

Comment:
This is a very long chain and seems quite brittle. I think it is going to break if it is not tested.

We could already remove one level by using sth like https://github.com/graalvm/truffleruby/blob/8a75511278048cf9806138ac28812b06f2ceb437/src/main/java/org/truffleruby/core/mutex/MutexOperations.java#L81-L87 in https://github.com/graalvm/truffleruby/blob/8a75511278048cf9806138ac28812b06f2ceb437/src/main/java/org/truffleruby/language/loader/FeatureLoader.java#L225 and make that a static method `rethrow` on say ExceptionOperations.